### PR TITLE
feat: Add build with artefact upload

### PR DESCRIPTION
**Pull request by indigo423**
Created a GitHub actions pipeline with an artifact upload to have something people can play with. The Google Protobuf was hardcoded to OSX x86_64 hardware. I've added a plugin that detects the architecture and allows it to build also on non-OSX systems.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+.DEFAULT_GOAL := compile
```
Comment (mvrueden) [inline Makefile:1]: -1 on `make`.

The project setup is pretty simple and there is no need for the `make` abstraction.

**Review by mvrueden — CHANGES_REQUESTED** [review]

**Comment by mvrueden** [comment]
FYI: The protobuf setup is temporarily. I generated the Java files manually without the maven plugin. In general it must be even more dynamic or OpenNMS should expose them properly. Hard-copy them over from OpenNMS source is not the way to go 😱

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+.PHONY: compile
+compile: deps
+	mvn compile
+
+.PHONY: tests
+tests:
+	mvn test
+
+.PHONY: build
+build: clean
```
Comment (fooker) [inline Makefile]:
```suggestion
build: compile
```

**Review by fooker — CHANGES_REQUESTED** [review]

**Comment by indigo423** [comment]
> FYI: The protobuf setup is temporarily. I generated the Java files manually without the maven plugin. In general it must be even more dynamic or OpenNMS should expose them properly. Hard-copy them over from OpenNMS source is not the way to go 😱

Thanks for the explanation, but what are the consequences of your explanation to the PR?

**Pull request closed without merging**